packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,52 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
```

Comment:
Nit: can we do something like this?

```suggestion
    if (this.matches(tt._with)) {
      this.expectPlugin("moduleAttributes");
      this.next(); // Consume the `with` keyword
    } else {
      if (this.hasPlugin("moduleAttributes")) return [];
      return null;
    }
```

By doing so, we throw the "missing plugin" error pointing at `with` rather than at what comes after.

Reply:
This makes sense. I will update the code 👍